### ISSUE TITLE
LSP :: No LSP server for cython-mode(check *lsp-log*).

### ISSUE THREAD

**Issue by yssource** [emacs-lsp/lsp-mode#1031]
**Describe the bug**
Since spacemacs hangs on, it is unable to view the `lsp-log`.
Please refer to https://github.com/syl20bnr/spacemacs/issues/12740

**Comment by yssource** [comment]
Thanks so much.

It still freezes the spacemacs, after the fix [9330d7c](https://github.com/emacs-lsp/lsp-mode/commit/9330d7c3f1431b38f84a2296572531f9574b7d6d) .

# Reproduce:
- `mkdir -v /tmp/hello && cd /tmp/hello && git init`
- `touch simpleclass.pxd`
- open simpleclass.pxd with spacemacs
- copy and paste, then save it in simpleclass.pxd with spacemacs

file: simpleclass.pxd
``` python
### distutils: language = c++

cdef extern from "SimpleClass.hpp":
    cdef cppclass Simple:
        Simple(int x)
        int get()
```

# Result:
CPU is running 100%, and spacemacs freezes.

**Comment by yssource** [comment]
This only happens to create, write and save a new `cython-mode` file.
There is no problem, if opening a existing `cython-mode` file.
